### PR TITLE
feat(edit-article):  limit to 10 connections per article

### LINKED ISSUE #3222: Update API for user to add up 10 connections per article
### Tasks
- [ ] Update API for allowing user to add up 10 connections per article
- [ ] Allow existing article to decrease the number
- [ ] Update test case

### References
- [Design](https://www.figma.com/file/vO8f8jaz9cZo1DsAazvHWd/Matters.news-%E8%BF%AD%E4%BB%A3%E8%AE%BE%E8%AE%A1?t=qez9FjTEBSH7jVT8-0)

### PR DESCRIPTION
**Pull request by gary02**
resolve #3222

### CODE REVIEW

File: src/common/enums/index.ts [hunk 1]
```diff
@@ -244,6 +244,8 @@ export enum ActivityType {
   UserAddArticleTagActivity = 'UserAddArticleTagActivity',
 }
 
+export const MAX_ARTICLES_PER_COLLECTION_LIMIT = 10
+
 export const MAX_TAGS_PER_ARTICLE_LIMIT = 5
```
Comment (pitb2022) [inline src/common/enums/index.ts:249]: MAX_COLLECTION_PER_ARTICLES__LIMIT ?

Comment (gary02) [reply src/common/enums/index.ts:249]: target article -> 1 collection ->  many related articles

File: src/mutations/article/editArticle.ts [hunk 2]
```diff
@@ -617,4 +514,145 @@ const resolver: MutationToEditArticleResolver = async (
...
+  knex,
+}: {
+  viewerId: string
+  collection: string[] | undefined | null
+  article: Article
+  dataSources: Pick<
+    DataSources,
+    'atomService' | 'userService' | 'articleService' | 'notificationService'
+  >
+  knex: any
```
Comment (tx0c) [inline src/mutations/article/editArticle.ts:536]: may import a `Knex` type and use it here?

File: src/mutations/article/editArticle.ts [hunk 3]
```diff
@@ -617,4 +514,145 @@ const resolver: MutationToEditArticleResolver = async (
...
+    })
+  ).map(({ articleId }: { articleId: string }) => articleId)
+  const newIds =
+    collection == null
+      ? []
+      : uniq(collection.map((articleId) => fromGlobalId(articleId).id)).filter(
+          (id) => !!id
+        )
+  const newIdsToAdd = difference(newIds, oldIds)
+  const oldIdsTodelete = difference(oldIds, newIds)
```
Comment (robertu7) [inline src/mutations/article/editArticle.ts:550]: oldIdsToDelete?

File: src/mutations/article/editArticle.ts [hunk 4]
```diff
@@ -617,4 +514,145 @@ const resolver: MutationToEditArticleResolver = async (
...
+
+  // do nothing if no change
+  if (isEqual(oldIds, newIds)) {
```
Comment (robertu7) [inline src/mutations/article/editArticle.ts:553]: have to sort `oldIds` and `newIds` before comparing?

**Review by robertu7 — APPROVED** [review]

**Pull request merged**